src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -336,4 +337,39 @@ private SearchRequest(
     public void setQuery(String q) {
         setQ(q);
     }
+
+    /**
+     * Method that returns the JSON String of the SearchRequest
+     *
+     * @return JSON String of the SearchRequest query
+     */
+    @Override
+    public String toString() {
+        JSONObject jsonObject =
+                new JSONObject()
+                        .put("q", this.q)
+                        .put("offset", this.offset)
+                        .put("limit", this.limit)
+                        .put("attributesToRetrieve", this.attributesToRetrieve)
+                        .put("cropLength", this.cropLength)
+                        .put("cropMarker", this.cropMarker)
+                        .put("highlightPreTag", this.highlightPreTag)
+                        .put("highlightPostTag", this.highlightPostTag)
+                        .put("matches", this.matches)
+                        .put("facetsDistribution", this.facetsDistribution)
+                        .put("sort", this.sort);
+        if (this.attributesToCrop != null) {
+            jsonObject.put("attributesToCrop", this.attributesToCrop);
```

Comment:
I think you can use `putOpt` [putOpt](https://developer.android.com/reference/org/json/JSONObject#putOpt(java.lang.String,%20java.lang.Object))

This way we can remove the if/else conditions here :)

Reply:
So cool